homemade_functions.py

```diff
@@ -0,0 +1,102 @@
+import random
+from random import randint
+from constants import*
+
+
+def convert_to_list(word):
+    deconstructed_word = []
+    for letter in word:
+        deconstructed_word.append(letter.upper())
+    return deconstructed_word
+
+
+def list_length(list_to_count):
+    count = 0
+    for i in list_to_count:
+        count += 1
+    return count
+
+def multiply_letter_pool(LETTER_POOL):
+    weighted_letters =[]
+    for letter, amount in LETTER_POOL.items():
+        grouped_letters = letter * amount
+        for letters in convert_to_list(grouped_letters):
+            weighted_letters.append(letters)
+    return weighted_letters
+
+
+def draw_hand():
+    final_letter_pool = multiply_letter_pool(LETTER_POOL)
+    pool_length = list_length(final_letter_pool)
+    drawn_letters = []
+    purgatory_hand = {}
+    drawn_letters_count = 0
+    while drawn_letters_count != 10:
```

Comment:
In Python, we call 10 in this context a "magic number". This means that the number has been hardcoded without much context. If I were a developer unfamiliar with Adagrams, I might not know exactly what that 10 means. We can avoid these magic numbers by making constants with more descriptive names to hold them. In this instance, we know that 10 is the max size of the hand, so we might do something along the lines of:

```suggestion
    MAX_HAND_SIZE = 10
    while drawn_letters_count != MAX_HAND_SIZE:
```

Reply:
oooh & it only has to do 2 passes per letter! cleaner _&_ more efficient!

Reply:
ChatGPT was right about me. I have to remember to not assume people know the backstory. taking the magic out of these numbers